app/src/main/java/com/example/cinemapp/ui/main/HomeFragment.kt

```diff
@@ -35,23 +36,30 @@ class HomeFragment : Fragment() {
 
     override fun onStart() {
         super.onStart()
+        setupViews()
 
         observeFlowSafely(viewModel.state) {
             adapter.setMovies(it.movies)
-            viewModel.togglePagingRunning()
+            viewModel.setPagingRunning(false)
         }
         binding.rvMovieList.addOnScrollListener(
             object : RecyclerView.OnScrollListener() {
                 override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
                     super.onScrolled(recyclerView, dx, dy)
                     if (binding.rvMovieList.isEndOfScroll()) {
-                        viewModel.getUpcomingNextPage()
+                        viewModel.getNextPage(viewModel.state.value.listType)
```

Comment:
We can avoid passing the parameter here, since the argument itself is inside `ViewModel`